langchain/agents/structured_chat/base.py

```diff
@@ -82,12 +83,13 @@ def create_prompt(
         tool_names = ", ".join([tool.name for tool in tools])
         format_instructions = format_instructions.format(tool_names=tool_names)
         template = "\n\n".join([prefix, formatted_tools, format_instructions, suffix])
+        if input_variables is None:
+            input_variables = ["input", "chat_history", "agent_scratchpad"]
         messages = [
             SystemMessagePromptTemplate.from_template(template),
+            MessagesPlaceholder(variable_name="chat_history"),
```

Comment:
why do we need this? The chat history is recorded in system and human with memory. I’m confused here.

Reply:
Nope, if you try to init this type of agent with a memory you will get an error, cause it’s currently zero shot